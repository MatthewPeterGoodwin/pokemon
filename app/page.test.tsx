import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Page from './page';

// Mock the components with the correct path relative to app directory
jest.mock('./components/PokemonSearch', () => {
  return function MockPokemonSearch() {
    return <div data-testid="pokemon-search">PokemonSearch Component</div>;
  };
});

jest.mock('./components/PokemonGallery', () => {
  return function MockPokemonGallery() {
    return <div data-testid="pokemon-gallery">PokemonGallery Component</div>;
  };
});

describe('Pokemon Home Page', () => {
  it('should render the page component', () => {
    render(<Page />);
    expect(screen.getByTestId('pokemon-search')).toBeInTheDocument();
    expect(screen.getByTestId('pokemon-gallery')).toBeInTheDocument();
  });

  it('should render components in correct order', () => {
    render(<Page />);
    
    const searchComponent = screen.getByTestId('pokemon-search');
    const galleryComponent = screen.getByTestId('pokemon-gallery');
    
    expect(searchComponent.compareDocumentPosition(galleryComponent))
      .toBe(Node.DOCUMENT_POSITION_FOLLOWING);
  });

  it('should display both components visibly', () => {
    render(<Page />);
    
    expect(screen.getByTestId('pokemon-search')).toBeVisible();
    expect(screen.getByTestId('pokemon-gallery')).toBeVisible();
  });

  it('should match snapshot', () => {
    const { container } = render(<Page />);
    expect(container).toMatchSnapshot();
  });
});