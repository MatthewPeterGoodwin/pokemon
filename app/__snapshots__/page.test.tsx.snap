// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Home Page should match snapshot 1`] = `
<div>
  <div
    data-testid="pokemon-search"
  >
    PokemonSearch Component
  </div>
  <div
    data-testid="pokemon-gallery"
  >
    PokemonGallery Component
  </div>
</div>
`;
